### PR TITLE
Support for older anaconda boot line options

### PR DESCRIPTION
**Pull request by xavierba**
Prior to RHEL 7 (and Fedora 17), anaconda was using different boot line options.
Cobbler has been (partially) switched to the newer options with 902d0c4. However, while this allows cobbler to work with anaconda versions that have removed the deprecated options (like CentOS Stream 9), this breaks anaconda versions that don't have the newer options, that is everything earlier than RHEL 7 and Fedora 17. All these distributions are indeed not supported anymore, but there are still use cases where installing these legacy distros is needed.

This adds checks for the distro os version in order to properly set the anaconda options.
It also takes care of the legacy 'kssendmac' which has been replaced by 'inst.ks.sendmac'.

### CODE REVIEW

**Review by olafhering — APPROVED** [review]
lgtm

**Comment by olafhering** [comment]
This was rebased, take the chance to remove the `v2` from the commit message and also the the MR text.

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
It would be nice if those new paths get tests. I don't care if integration or unit but I would kindly ask that we satisfy Codecov.

**Comment by xavierba** [comment]
@SchoolGuy I did not had time to look at tests yet, sorry :-(

**Comment by SchoolGuy** [comment]
@xavierba No worries. Thanks for the update. If possible I would like to get this merged in the next two weeks, so we can slowly start getting things in for the 3.3.1 release.

**Comment by SchoolGuy** [comment]
@xavierba Can you take time and add tests in the near future? Also we added some additional tests, so maybe a rebase onto `master` already helps a bit.

**Comment by xavierba** [comment]
@SchoolGuy about looking at that, sorry for the delay, again... Do you have any pointers ? Is that ok to duplicate the Arrange/Act/Assert block in test_generate_system/test_generate_profile from tests/actions/buildiso_test.py and tune it to test for, say rhel6 and rhel7 ? Not a developer and my cobbler test install is running 3.2, not 3.3/master.

**Comment by SchoolGuy** [comment]
@xavierba Yes for all of the mentioned things are fine. Sorry for not seeing your ping. I don't know how I was able to miss it.

**Comment by xavierba** [comment]
@SchoolGuy sorry for using an incorrect base branch. Is that better now with the added tests commit ?

File: tests/actions/buildiso_test.py [hunk 1]
```diff
@@ -311,3 +344,57 @@ def test_generate_profile(
...
+        # Very basic test yes but this is the expected result atm
+        # TODO: Make tests more sophisticated
+        assert (
+            result
+            == " append initrd=testdistro.img inst.ks=default.ks"
+        )
+
+        # Arrange
+        folder = create_kernel_initrd(fk_kernel, fk_initrd)
+        kernel_path = os.path.join(folder, fk_kernel)
```
Comment (SchoolGuy) [inline tests/actions/buildiso_test.py:377]: I am confused by this bit? Is there a function header missing or is this done on purpose?

File: tests/actions/buildiso_test.py [hunk 2]
```diff
@@ -311,3 +344,57 @@ def test_generate_profile(
             result
             == " append initrd=testdistro.img install=http://192.168.1.1:80/cblr/links/testdistro autoyast=default.ks"
         )
+
+        # Arrange
+        folder = create_kernel_initrd(fk_kernel, fk_initrd)
```
Comment (SchoolGuy) [inline tests/actions/buildiso_test.py:349]: Same question here

File: tests/actions/buildiso_test.py (hunk 1 above)
Comment (xavierba) [reply tests/actions/buildiso_test.py:377]: This is done on purpose, but from your comment, I understand this is not the way it should be done, or at least not the way you expected it to be done.

Comment (SchoolGuy) [reply tests/actions/buildiso_test.py:377]: My idea is to have small independent tests. Each test has one arrange, one act and one assert stage. Cleanup should be done in pytest fixtures. Arrange may be a fixture but doesn't have to be. Please try to reuse existing fixtures to reduce the amount of testcode.

Comment (SchoolGuy) [reply tests/actions/buildiso_test.py:377]: @xavierba Any chance we can get my wish in this PR? I would really love to have it.

**Comment by tpw56j** [comment]
Why not just tweak the `/etc/cobbler/boot_loader_conf/{pxe|ipxe|grub}.template`?
For example my `pxe.template`:
````diff
@@ -23,6 +23,14 @@
 	localboot -1
 #else
 	kernel $kernel_path
+#if $breed == "redhat"
+#if $os_version == "rhel6"
+#set $append_line=$append_line.replace("inst.", "") + " ksdevice=bootif lang=en text"
+#else
+#set $append_line=$append_line.replace(" ks=", " inst.ks=")
+#set $append_line=$append_line.replace("kssendmac", "inst.ks.sendmac") + " inst.ks.device=bootif inst.lang=en inst.text inst.ks.sendsn"
+#end if
+#end if
 #if $breed != "windows"
 	$append_line
 	ipappend 2
````

**Comment by tpw56j** [comment]
Sorry, transferring complexity and opacity from code to templates was not a good idea. Maybe move the boot line options to `distro_signatures.json` and completely clean up the code and templates from "if breed" and "if os_version"?
For example:
````json
      "rhel6": {
        "kernel_options": "kssendmac ks={autoinstall}",
        "kernel_options_standalone": "ks=cdrom:/isolinux/{descendant}.cfg",
      },
````
````json
      "rhel9": {
        "kernel_options": "inst.ks.sendmac inst.ks={autoinstall}",
        "kernel_options_standalone": "inst.ks=cdrom:/isolinux/{descendant}.cfg",
      },
````

**Comment by SchoolGuy** [comment]
@tpw56j That would work nicely in only some cases. In most cases that only provides a default. And since these options should be handled by the server, this should really not be "visible" to the user at that level.

**Comment by SchoolGuy** [comment]
This is superseded by #3335

**Pull request closed without merging**